port/unix/omrsysinfo.c

```diff
@@ -3617,6 +3655,63 @@ readCgroupSubsystemFile(struct OMRPortLibrary *portLibrary, uint64_t subsystemFl
 	return rc;
 }
 
+/** 
+ * Checks if the process is running inside docker container 
+ *
+ * @param[in] portLibrary pointer to OMRPortLibrary
+ * @param[out] inDocker pointer to BOOLEAN which on successful return indicates if the process is running in docker container or not 
+ *
+ * @return 0 on success, otherwise negative error code
+ */
+static int32_t
+isRunningInDocker(struct OMRPortLibrary *portLibrary, BOOLEAN *inDocker)
+{
+	int32_t rc = 0;
+
+	/* Assume we are not in docker*/
+	*inDocker = FALSE;
+
+	if (isCgroupV1Available(portLibrary)) {
+		/* Read PID 1's cgroup file /proc/1/cgroup and check cgroup name for each subsystem.
+		 * If cgroup name for each subsystem points to the root cgroup "/",
+		 * then the process is not running in a container.
+		 * For any other cgroup name, assume we are in a 'docker' container.
+		 */
+		FILE *cgroupFile = fopen("/proc/1/cgroup", "r");
+
+		if (NULL == cgroupFile) {
+			rc = portLibrary->error_set_last_error(portLibrary, errno, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_FOPEN_FAILED);
+			goto _end;
+		}
+
+		while (0 == feof(cgroupFile)) {
+			char cgroup[PATH_MAX];
+			char subsystems[1024];
+			int32_t hierId = -1;
+
+			rc = fscanf(cgroupFile, PROC_PID_CGROUP_ENTRY_FORMAT, &hierId, subsystems, cgroup);
+			/* Ensure we didn't overflow */
+			Assert_PRT_true(strlen(subsystems) < 1024);
+			Assert_PRT_true(strlen(cgroup) < PATH_MAX);
+
+			if (EOF == rc) {
+				break;
+			} else if (3 != rc) {
+				rc = portLibrary->error_set_last_error_with_message_format(portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_READ_FAILED, "unexpected format of /proc/1/cgroup file");
+				goto _end;
```

Comment:
`@return 0 on success, otherwise negative error code`

This may exit the method with a positive, non-zero return value contrary to the function contract.

Reply:
`rc` is reassigned return value of `error_set_last_error_with_message_format` which is negative error code.

Reply:
I missed that - sorry.